moshi/src/main/java/com/squareup/moshi/Moshi.java

```diff
@@ -171,6 +172,12 @@ public Builder add(Object adapter) {
       return add(AdapterMethodsFactory.get(adapter));
     }
 
+    /** Add all factories enclosed in the provided moshi instance. */
+    public Builder add(Moshi moshi) {
```

Comment:
I think we probably want `newBuilder()` instead of this. As-is you can compose a `Moshi` from multiple other `Moshi` instances, and that doesn't make much sense.